Automattic/wp-calypso
# Move lib/posts files to state/posts and convert editor analytics to Redux

This PR is very big, but it's a mostly mechanical move of the remaining `lib/posts` files (stats, actions, utils) to `state/posts`.

The changes that don't just move code are:

**Chai `expect` asserts in `utils` tests**
I converted the `expect` asserts in `state/posts/test/utils` from Chai to Jest. When merging the two `utils` test suites into one file (from `lib` into the existing `state`) there was a conflict where one file used Chai `expect` asserts while the other one was already Jest.

**Stats recording functions converted to Redux actions**
In `state/posts/stats`, the `recordEvent` and `recordStat` functions were converted to `recordEditorEvent` and `recordEditorStat` action creators and the usages were replaced by Redux action dispatches.

This removed direct import of `lib/analytics` that depends on DOM and BOM and breaks tests. The Redux analytics middleware is now used for most Post Editor analytics now.

**How to test:**
Enable the analytics debug logging (`localStorage.debug = 'calypso:analytics:*'` in console) and work with the editor. Verify that the correct analytics are fired -- there are plenty of them on almost every click.
